File: 05-searching/01-binary-search-in-sorted-array/01-binary-search-in-sorted-array.rb
def binary_search_in_sorted_array(numbers, key)
  left, right = 0, numbers.length-1
  while left <= right 
    mid = (left+right)/2
    return true if numbers[mid] == key
    key < numbers[mid] ? right = mid-1 : left = mid+1
  end
  return false
end

puts binary_search_in_sorted_array([10, 20, 30, 40, 50, 60], 20)

puts binary_search_in_sorted_array([10, 20, 30, 40, 50, 60], 70)

File: 05-searching/01-binary-search-in-sorted-array/01-binary-search-in-sorted-array.ts
const binary_search_in_sorted_array = (numbers: number[], key: number) => {
  let left = 0
  let right = numbers.length - 1
  while (left <= right) {
    const mid = Math.floor((left + right) / 2)
    if (numbers[mid] === key) return true
    else [left, right] = key < numbers[mid] ? [left, mid - 1] : [mid + 1, right]
  }
  return false
}

console.log(binary_search_in_sorted_array([10, 20, 30, 40, 50, 60], 20))

console.log(binary_search_in_sorted_array([10, 20, 30, 40, 50, 60], 70))
